src/rust/src/expr/string.rs

```diff
@@ -0,0 +1,395 @@
+            .str()
+            .extract_groups(pattern)
+            .map_err(RPolarsErr::from)?
+            .into())
+    }
+
+    fn str_count_matches(&self, pat: &PlRExpr, literal: bool) -> Result<Self> {
+        Ok(self
+            .inner
+            .clone()
+            .str()
+            .count_matches(pat.inner.clone(), literal)
+            .into())
+    }
+
+    // fn str_to_date(&self, format: &str, strict: bool, exact: bool, cache: bool) -> Result<Self> {
+    //     let format = format.to_string().into();
+    //     Ok(self
+    //         .inner
+    //         .clone()
+    //         .str()
+    //         .to_date(StrptimeOptions {
+    //             format,
+    //             strict,
+    //             exact,
+    //             cache,
+    //         })
+    //         .into())
+    // }
+
+    // #[allow(clippy::too_many_arguments)]
+    // fn str_to_datetime(
+    //     &self,
+    //     format: &PlRExpr,
+    //     time_unit: &PlRExpr,
+    //     time_zone: &PlRExpr,
+    //     strict: &PlRExpr,
+    //     exact: &PlRExpr,
+    //     cache: &PlRExpr,
+    //     ambiguous: &PlRExpr,
+    // ) -> Result<Self> {
+    //     let format = robj_to!(Option, String, format)?.map(|x| x.into());
+    //     let time_unit = robj_to!(Option, timeunit, time_unit)?.map(|x| x.into());
+    //     let time_zone = robj_to!(Option, String, time_zone)?.map(|x| x.into());
+
+    //     Ok(self
+    //         .inner
+    //         .clone()
+    //         .str()
+    //         .to_datetime(
+    //             time_unit,
+    //             time_zone,
+    //             StrptimeOptions {
+    //                 format: format,
+    //                 strict: robj_to!(bool, strict)?,
+    //                 exact: robj_to!(bool, exact)?,
+    //                 cache: robj_to!(bool, cache)?,
+    //             },
+    //             robj_to!(PLExpr, ambiguous)?,
+    //         )
+    //         .into())
+    // }
+
+    // fn str_to_time(&self, format: &PlRExpr, strict: &PlRExpr, cache: &PlRExpr) -> Result<Self> {
+    //     let format = robj_to!(Option, String, format)?.map(|x| x.into());
+
+    //     Ok(self
+    //         .inner
+    //         .clone()
+    //         .str()
+    //         .to_time(StrptimeOptions {
+    //             format,
+    //             strict: robj_to!(bool, strict)?,
+    //             cache: robj_to!(bool, cache)?,
+    //             exact: true,
+    //         })
+    //         .into())
+    // }
+
+    fn str_split(&self, by: &PlRExpr, inclusive: bool) -> Result<Self> {
+        if inclusive {
+            Ok(self
+                .inner
+                .clone()
+                .str()
+                .split_inclusive(by.inner.clone())
+                .into())
+        } else {
+            Ok(self.inner.clone().str().split(by.inner.clone()).into())
+        }
+    }
+
+    fn str_split_exact(&self, by: &PlRExpr, n: usize, inclusive: bool) -> Result<Self> {
```

Comment:
I didn't know that savvy supports usize, I looked at the savvy implementation and it seems to only support conversion from i32.
So it would be better to do the conversion from NumericScalar via `<Wrap<usize>>::try_from` for now.

@yutannihilation If savvy supports usize input, do you think it would be worth adding the following conversion?
https://github.com/eitsupi/neo-r-polars/blob/b90688621f7d6c4e35e34574421bc63122c58f35/src/rust/src/conversion/mod.rs#L202-L227

Reply:
Hmm..., I might change my mind, but no plan. `f64` and `i32` are supported because these are the actual memory representation used by R, but there's no corresponding type to `usize` in R.

Reply:
Thanks for your comment.
In that case it seems quite strange that usize is supported there.
Why is usize supported as an scalar input?
I didn't know it existed because it wasn't mentioned in the book

Reply:
Oh..., no idea. Probably I thought it would be useful, then forgot it. So, I should remove it for consistency?

> it wasn't mentioned in the book

This is a good point... An exception makes documentation a bit complex both for readers and for authors.

Reply:
> So, I should remove it for consistency?

If it is clear why only usize is supported, I think there is no need to remove it.
Why usize is there but not u32 or u64 is odd at first glance.

Reply:
An integer can be intended as an index. So, usize is supported. u32 and u64 is not the case.

Reply:
In that case, is there any reason not to allow conversion from R double to usize, as I suspect integer32 is too small to cover the full range of usize.

Reply:
This is just for convenience for casual usages. An ordinary R user doesn't index a vector longer than 2^32.

Reply:
Ok, thanks.
In any case, it would be great if it were documented.